lib/xnvme_be_spdk_dev.c

```diff
@@ -689,6 +696,55 @@ xnvme_be_spdk_enumerate(const char *sys_uri, struct xnvme_opts *opts, xnvme_enum
 	return 0;
 }
 
+static int
+reconnect_ctrlr(struct spdk_nvme_ctrlr *ctrlr)
+{
+	int err;
+	XNVME_DEBUG("WARNING: attempting to reconnect controller");
+	err = spdk_nvme_ctrlr_reset(ctrlr);
+	if (err < 0) {
+		XNVME_DEBUG("FAILED: spdk_nvme_ctrlr_reset, err: %d", err);
+		return -EBUSY;
+	}
+	XNVME_DEBUG("INFO: controller reconnected");
+	return 0;
+}
+
+void
+_keep_alive_cb(void *ctx, const struct spdk_nvme_cpl *cpl)
+{
+	int *result = (int *)ctx;
+	*result = cpl->status.sct;
```

Comment:
This should be `*result = cpl->status.sc;` instead, right now we are checking the status code type instead of the status code.